vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/IntegerField.java

```diff
@@ -31,7 +32,8 @@
 @JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
 @NpmPackage(value = "@vaadin/integer-field", version = "24.0.0-alpha6")
 @JsModule("@vaadin/integer-field/src/vaadin-integer-field.js")
-public class IntegerField extends AbstractNumberField<IntegerField, Integer> {
+public class IntegerField extends AbstractNumberField<IntegerField, Integer>
+        implements HasThemeVariant<TextFieldVariant> {
```

Comment:
`IntegerField` does not implement `HasAllowedCharPattern` anymore. Intentional?

Reply:
Yes, because it is [hardcoded](https://github.com/vaadin/web-components/blob/ee7af5454f325a64c892d4a7ddffc268faaa7472/packages/integer-field/src/vaadin-integer-field.js#L45) in the web component and not supposed to be changed by the user.
Same applies to `BigDecimalField` which also doesn't implement `HasAllowedCharPattern`.